Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraint.cs

```diff
@@ -545,6 +587,38 @@ private static Ray CalculateProjectedSafeZoneRay(Vector3 origin, Transform targe
             return new Ray(origin + direction, -direction);
         }
 
+        private static Ray CalculateProjectedSafeZoneRay(
+            Vector3 origin, Transform targetTransform, IMixedRealityController hand,
+            SolverSafeZone handSafeZone, SolverOffsetBehavior offsetBehavior, float angleOffset)
+        {
+            // Ignore the angle offset if safe zone is atop palm, does not apply in that case.
+            if (angleOffset == 0 || handSafeZone == SolverSafeZone.AtopPalm)
+            {
+                return CalculateProjectedSafeZoneRay(origin, targetTransform, hand, handSafeZone, offsetBehavior);
+            }
+
+            angleOffset = angleOffset % 360;
+            while (angleOffset < 0)
+            {
+                angleOffset = (angleOffset + 360) % 360;
+            }
+
+            var offset = angleOffset / 90;
+            var intOffset = Mathf.FloorToInt(offset);
+            var fracOffset = offset - intOffset;
+
+            var currentSafeZoneClockwiseIdx = System.Array.IndexOf(handSafeZonesClockWiseRightHand, handSafeZone);
+
+            var intPartSafeZoneClockwise = handSafeZonesClockWiseRightHand[(currentSafeZoneClockwiseIdx + intOffset) % SolverSafeZoneClockwiseCount];
```

Comment:
A bit of a nit, but it'd be super useful to show the type throughout this method instead of `var`. I'm finding myself scrolling up and down to figure out what types methods return or this array contains.

Reply:
Done!